### PR TITLE
[WIP] Refactor common remotecommand code into apimachinery from client-go

### PR DESCRIPTION
**Pull request by seans3**
* Refactor common `remotecommand` code from `client-go` to `apimachinery`.
  * Creates aliases for exported symbols in former `client-go` package pointing to new `apimachinery` symbols.
  * Moves `StreamTypes` from `k8s.io/api/core/v1` to `k8s.io/apimachinery/pkg/util/httpstream` in order to be able to use these types in `apimachinery`.
  * Moves `ExitError` and `CodeExitError` into more appropriate `k8s.io/apimachinery/pkg/api/errors` from `k8s.io/client-go/util/exec`.
  * Updates imports to use new `apimachinery` package instead of older `client-go`.
* Helps with upcoming transition from SPDY to WebSockets
  * Upcoming `StreamTranslator` proxy in `apimachinery` will need SPDY client code in `apimachinery`, since the WebSocket communication leg will only traverse the client to the `APIServer`. The rest of the streaming communication from the `APIServer` through the kubelet and the container runtime will continue using SPDY. WebSockets will eventually traverse all the way to the container, but the first step is just the client -> `APIServer` leg.

/kind cleanup

```release-note
NONE
```

- [KEP Transition from SPDY to WebSockets - 4006](https://github.com/kubernetes/enhancements/issues/4006)

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/exec/exec.go [hunk 1]
```diff
@@ -27,12 +27,13 @@ import (
...
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/remotecommand"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/cli-runtime/pkg/genericiooptions"
 	"k8s.io/cli-runtime/pkg/resource"
 	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
 	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/remotecommand"
+	remoteclient "k8s.io/client-go/tools/remotecommand"
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/exec/exec.go:36]: Nice to see such distinguishable renaming(remoteclient, remoteserver, remotecommand), it makes following code easier.

File: staging/src/k8s.io/kubectl/pkg/util/term/resizeevents.go [hunk 2]
```diff
@@ -24,8 +24,8 @@ import (
 	"os/signal"
 
 	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/util/remotecommand"
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/util/term/resizeevents.go:27]: nit: new line after `golang.org/x/sys/unix` would be preferable

**Review by ardaguclu — COMMENTED** [review]
/lgtm

**Comment by seans3** [comment]
Abandoning this refactor. This refactor is awkward moving SPDY client code into `apimachinery`.  WebSocket transition which requires SPDY client code in `apimachinery` will copy needed code instead.

**Pull request closed without merging**